local_planner/src/nodes/avoidance_output.h

```diff
@@ -11,29 +11,36 @@ enum waypoint_choice { hover, costmap, tryPath, direct, reachHeight, goBack };
 
 struct avoidanceOutput {
   waypoint_choice waypoint_type;
-
-  geometry_msgs::PoseStamped pose;
-  bool obstacle_ahead;
-  bool reach_altitude;
-  double min_speed;
-  double max_speed;
-  double velocity_sigmoid_slope;
-  ros::Time last_path_time;
-
-  bool use_avoid_sphere;
-  int avoid_sphere_age;
-  geometry_msgs::Point avoid_centerpoint;
-  double avoid_radius;
-
-  geometry_msgs::Point back_off_point;
-  geometry_msgs::Point back_off_start_point;
-  double min_dist_backoff;
-
-  geometry_msgs::PoseStamped take_off_pose;
-  geometry_msgs::PoseStamped offboard_pose;
-
-  double costmap_direction_e;
-  double costmap_direction_z;
-  std::vector<geometry_msgs::Point> path_node_positions;
+  geometry_msgs::PoseStamped pose;  // current vehicle position
+  bool obstacle_ahead;  // true is there is an obstacle ahead of the vehicle
+  bool reach_altitude;  // true if the vehicle has reach the goal altitude
+  double min_speed;     // minimum speed
```

Comment:
do we need double precision for these? Same in other places. If the parameters are intrinsically doubles, then sure, but I don't think this is the case?

Reply:
yes, ROS parameters and ros messages like PoseStamped, Point are double. 
From the ROS wiki
- paramtype - defines the type of value stored, and can be any of int_t, double_t, str_t, or bool_t from http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
- 

Primitive Type | Serialization | C++ | Python2 | Python3
-- | -- | -- | -- | --
float64 | 64-bit IEEE float | double | float | float



from http://wiki.ros.org/msg 

I have started converting some double into float here and in #199 little by little.

Reply:
ok, that's what I figured. All good!